<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>LaScaMolMR · LaScaMolMR.jl</title><meta name="title" content="LaScaMolMR · LaScaMolMR.jl"/><meta property="og:title" content="LaScaMolMR · LaScaMolMR.jl"/><meta property="twitter:title" content="LaScaMolMR · LaScaMolMR.jl"/><meta name="description" content="Documentation for LaScaMolMR.jl."/><meta property="og:description" content="Documentation for LaScaMolMR.jl."/><meta property="twitter:description" content="Documentation for LaScaMolMR.jl."/><script data-outdated-warner src="assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.050/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.8/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="assets/documenter.js"></script><script src="search_index.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/catppuccin-mocha.css" data-theme-name="catppuccin-mocha"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/catppuccin-macchiato.css" data-theme-name="catppuccin-macchiato"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/catppuccin-frappe.css" data-theme-name="catppuccin-frappe"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/catppuccin-latte.css" data-theme-name="catppuccin-latte"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><a class="docs-logo" href="index.html"><img src="assets/logo.png" alt="LaScaMolMR.jl logo"/></a><div class="docs-package-name"><span class="docs-autofit"><a href="index.html">LaScaMolMR.jl</a></span></div><button class="docs-search-query input is-rounded is-small is-clickable my-2 mx-auto py-1 px-2" id="documenter-search-query">Search docs (Ctrl + /)</button><ul class="docs-menu"><li class="is-active"><a class="tocitem" href="index.html">LaScaMolMR</a><ul class="internal"><li><a class="tocitem" href="#Overview"><span>Overview</span></a></li><li><a class="tocitem" href="#Tutorial"><span>Tutorial</span></a></li><li><a class="tocitem" href="#Contents"><span>Contents</span></a></li><li><a class="tocitem" href="#API"><span>API</span></a></li><li><a class="tocitem" href="#Index"><span>Index</span></a></li></ul></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><a class="docs-sidebar-button docs-navbar-link fa-solid fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href="index.html">LaScaMolMR</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href="index.html">LaScaMolMR</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl" title="View the repository on GitHub"><span class="docs-icon fa-brands"></span><span class="docs-label is-hidden-touch">GitHub</span></a><a class="docs-navbar-link" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/main/docs/src/index.md" title="Edit source on GitHub"><span class="docs-icon fa-solid"></span></a><a class="docs-settings-button docs-navbar-link fa-solid fa-gear" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-article-toggle-button fa-solid fa-chevron-up" id="documenter-article-toggle-button" href="javascript:;" title="Collapse all docstrings"></a></div></header><article class="content" id="documenter-page"><h1 id="LaScaMolMR"><a class="docs-heading-anchor" href="#LaScaMolMR">LaScaMolMR</a><a id="LaScaMolMR-1"></a><a class="docs-heading-anchor-permalink" href="#LaScaMolMR" title="Permalink"></a></h1><h2 id="Overview"><a class="docs-heading-anchor" href="#Overview">Overview</a><a id="Overview-1"></a><a class="docs-heading-anchor-permalink" href="#Overview" title="Permalink"></a></h2><p>LaScaMolMR.jl (Large Scale Molecular Mendelian Randomization) is a threaded Mendelian Randomization (MR) package that is focused on the generation of transcriptome wide / molecular MR analysies. Although it provides an interface for most common MR regression estimators (Inverse Variance Weighted, Weighted Median, Egger, Wald), its intended use is to enable fast Omic-wide Mendelian Randomization studies. The rise of large genetic cohort data has benefited the statistical power of Genome Wide Association Studies (GWAS) and Quantitative Trait Loci (QTL). Thus enabling findings in extensive studies such as Transcriptome Wide MR (TWMR). LaScaMolMR.jl provides a fast and efficient framework to such analyses, allowing users to customize the parameters of the study.</p><h2 id="Tutorial"><a class="docs-heading-anchor" href="#Tutorial">Tutorial</a><a id="Tutorial-1"></a><a class="docs-heading-anchor-permalink" href="#Tutorial" title="Permalink"></a></h2><h3 id="Example-1-:-Cis-MR"><a class="docs-heading-anchor" href="#Example-1-:-Cis-MR">Example 1 : Cis-MR</a><a id="Example-1-:-Cis-MR-1"></a><a class="docs-heading-anchor-permalink" href="#Example-1-:-Cis-MR" title="Permalink"></a></h3><p>For a QTL dataset composed as follows :</p><pre><code class="nohighlight hljs">base/folder
├── exposureA
│   ├── exposureA_chr1.txt
│   └── exposureA_chr2.txt
├── exposureB
│   ├── exposureB_chr1.txt
│   └── exposureB_chr2.txt
└── exposureC
    ├── exposureC_chr3.txt
    └── exposureC_chr4.txt</code></pre><p>With example data composed like this (tab separated) :</p><pre><code class="nohighlight hljs">chr	pos	effect_allele	other_allele	beta	se	some_other_column pval
1	10511	A	G	-0.176656	0.136297	.   0.194939
1	10642	A	G	-0.724554	0.345390	..  0.035924
1	11008	G	C	-0.017786	0.016673	... 0.286088
1	11012	G	C	-0.017786	0.016673	..  0.286088
1	13110	A	G	0.013272	0.021949	.   0.545411
1	13116	G	T	-0.027802	0.013111	..  0.0339672
1	13118	G	A	-0.027802	0.013111	... 0.0339672
1	13259	A	G	-0.122207	0.210776	..  0.562052
1	13273	C	G	0.007077	0.015337	.   0.644463</code></pre><p>and a GWAS of outcome composed similarly but comma separated, the following code will generate a cis-MR study with default parameters :</p><pre><code class="language-julia hljs">using LaScaMolMR</code></pre><ol><li>Describe the exposure data structure (files and columns).</li></ol><pre><code class="language-julia hljs">path_pattern = [&quot;exposure&quot;, TRAIT_NAME, &quot;/exposure&quot;, TRAIT_NAME, &quot;_chr&quot;, CHR, &quot;.txt&quot;]
columns = Dict(1 =&gt; CHR, 2 =&gt; POS, 3 =&gt; A_EFFECT, 4 =&gt; A_OTHER, 5 =&gt; BETA, 6 =&gt; SE, 8 =&gt; PVAL)

trait_v = [&quot;A&quot;, &quot;B&quot;, &quot;C&quot;] # exposure trait identifiers
chr_v = [1, 2, 3] # chromosome corresponding to exposure protein or gene
tss_v = [45287, 984276, 485327765] # position of TSS, if relevant

exposure::QTLStudy = QTLStudy_from_pattern(&quot;base/folder/&quot;, 
                                            path_pattern, 
                                            trait_v, chr_v, tss_v, 
                                            columns, separator = &#39;\t&#39;)</code></pre><ol><li>Describe the outcome data.</li></ol><pre><code class="language-julia hljs">outcome = GWAS(&quot;/some/file&quot;, columns, separator = &#39;,&#39;, trait_name = &quot;Some Painful Disease&quot;)</code></pre><ol><li>Perform Medelian Randomization study by providing input formats and reference genotype data files.</li></ol><pre><code class="language-julia hljs"># Plink 1.9 files base names for each chromosome (You can also use a single file)
plink_files = [&quot;folder/basename_file_chr$(i)&quot; for i in 1:22]

# Perform MR for every exposure - outcome pairs with default parameters
out_table = mrStudyCis(exposure, outcome, plink_files)

# with MiLoP approach to internal pleiotropy and other parameters :
out_table_MiLoP = mrStudyCis(exposure, outcome, plink_files, 
                        approach = &quot;MiLoP&quot;, 
                        r2_tresh = 0.01, 
                        p_tresh = 5e-8,
                        p_tresh_MiLoP = 5e-4,
                        pval_bigfolat = true)</code></pre><h3 id="Example-2-:-Trans-MR"><a class="docs-heading-anchor" href="#Example-2-:-Trans-MR">Example 2 : Trans-MR</a><a id="Example-2-:-Trans-MR-1"></a><a class="docs-heading-anchor-permalink" href="#Example-2-:-Trans-MR" title="Permalink"></a></h3><p>For a QTL dataset composed as follows :</p><pre><code class="nohighlight hljs">base/folder
├── exposureA.txt
├── exposureB.txt
└── exposureC.txt</code></pre><p>With example data composed like this (tab separated) :</p><pre><code class="nohighlight hljs">chr	pos	A1	A2	beta	se	some_useless_column pval
1	10511	A	G	-0.176656	0.136297	.   0.194939
1	10642	A	G	-0.724554	0.345390	..  0.035924
1	11008	G	C	-0.017786	0.016673	... 0.286088
1	11012	G	C	-0.017786	0.016673	..  0.286088
1	13110	A	G	0.013272	0.021949	.   0.545411
1	13116	G	T	-0.027802	0.013111	..  0.0339672
1	13118	G	A	-0.027802	0.013111	... 0.0339672
1	13259	A	G	-0.122207	0.210776	..  0.562052
1	13273	C	G	0.007077	0.015337	.   0.644463</code></pre><p>and a GWAS of outcome composed similarly but comma separated, the following code will generate a trans-MR study with default parameters :</p><pre><code class="language-julia hljs">using LaScaMolMR</code></pre><ol><li>Describe exposure data.</li></ol><pre><code class="language-julia hljs">path_pattern = [&quot;exposure&quot;, TRAIT_NAME, &quot;.txt&quot;]
columns = Dict(1 =&gt; CHR, 2 =&gt; POS, 3 =&gt; A_EFFECT, 4 =&gt; A_OTHER, 5 =&gt; BETA, 6 =&gt; SE, 8 =&gt; PVAL)

trait_v = [&quot;A&quot;, &quot;B&quot;, &quot;C&quot;] # Chromosome and TSS informations are not relevant in Trans setting.

exposure::QTLStudy = QTLStudy_from_pattern(&quot;base/folder/&quot;, 
                                            path_pattern, 
                                            trait_v, chr_v = nothing, 
                                            tss_v = nothing, 
                                            columns, separator = &#39;\t&#39;, 
                                            only_corresp_chr = false)</code></pre><ol><li>Describe outcome data.</li></ol><pre><code class="language-julia hljs">outcome = GWAS(&quot;/some/file&quot;, columns, separator = &#39;,&#39;, trait_name = &quot;Some Painful Disease&quot;)</code></pre><ol><li>Perform Medelian Randomization study by providing input formats and reference genotype data files.</li></ol><pre><code class="language-julia hljs"># Plink 1.9 files base names for each chromosome (You can also use a single file)
plink_files = [&quot;folder/basename_file_chr$(i)&quot; for i in 1:22]

# Perform MR for every exposure - outcme pairs with default parameters
out_table = mrStudyTrans(exposure, outcome, plink_files)

# with MiLoP approach and other parameters :
out_table2 = mrStudyTrans(exposure, outcome, plink_files, 
                        approach = &quot;MiLoP&quot;, 
                        r2_tresh = 0.01, 
                        p_tresh = 5e-8,
                        filter_beta_ratio = 1)
# Default p_tresh_MiLoP value is the same as p_tresh</code></pre><h3 id="MiLoP-approach"><a class="docs-heading-anchor" href="#MiLoP-approach">MiLoP approach</a><a id="MiLoP-approach-1"></a><a class="docs-heading-anchor-permalink" href="#MiLoP-approach" title="Permalink"></a></h3><p>Mitigated Local Pleiotropy (MiLoP) approach modifies the potential IV selection process to remove instrument variables associated to more than 1 exposure at <code>p_tresh_MiLoP</code> significance level.</p><h3 id="mr_*-regression-functions"><a class="docs-heading-anchor" href="#mr_*-regression-functions"><code>mr_*</code> regression functions</a><a id="mr_*-regression-functions-1"></a><a class="docs-heading-anchor-permalink" href="#mr_*-regression-functions" title="Permalink"></a></h3><p>LaScaMolMR provides four MR regression functions : <code>mr_wald</code>, <code>mr_ivw</code>, <code>mr_egger</code> and <code>mr_wm</code> performing respectively the wald ratio of the first provided instrument, Inverse Variance Weighted, Egger and Weighted Median regressions.</p><p>Example :</p><pre><code class="language-julia hljs">out = mr_wm(beta_outcome, se_outcome, beta_exposure, se_exposure;
            iteration = 10000, seed = 42, α = 0.05)</code></pre><h3 id="mr_output-as-a-standard-output-for-mr_*-:"><a class="docs-heading-anchor" href="#mr_output-as-a-standard-output-for-mr_*-:"><code>mr_output</code> as a standard output for <code>mr_*</code> :</a><a id="mr_output-as-a-standard-output-for-mr_*-:-1"></a><a class="docs-heading-anchor-permalink" href="#mr_output-as-a-standard-output-for-mr_*-:" title="Permalink"></a></h3><p>The mr<em>output struct provies a standard interface to functions performing mendelian randomization. This allows for user to use its own MR functions for <code>mrStudy</code>. Any function receiving 4 vectors, having at least <span>$\alpha$</span> as an option and outputing a `mr</em>output` is valid.</p><p>Here is the definitioin of <code>mr_output</code> :</p><pre><code class="language-julia hljs">struct mr_output
    nivs::Int
    effect::Float64
    se_effect::Float64
    ci_low::Float64
    ci_high::Float64
    p::Float64
    intercept::Float64
    p_intercept::Float64
    ci_low_intercept::Float64
    ci_high_intercept::Float64
    heter_stat::Float64
    heter_p::Float64
end</code></pre><p>Any function following this format could be provided to <a href="index.html#LaScaMolMR.ClumpAndMR"><code>ClumpAndMR</code></a>/<a href="index.html#LaScaMolMR.mrStudy"><code>mrStudy</code></a> in the <code>mr_methods</code> option, including user-defined functions.</p><pre><code class="language-julia hljs">function mr_something(beta_outcome, se_outcome, beta_exposure, se_exposure; 
                      α = 0.05, other_params = default_value
                     )::mr_output

    # Calculate values
    ...
    # end

    # Not calculated values can be replaced by NaN
    return mr_ouput(nivs, effect, ci_low, ci_high, p, NaN, NaN, NaN, NaN, heter_stat, heter_p)

end

output = NaiveCis(data, genotypes; mr_methods = [mr_ivw, mr_something])</code></pre><h2 id="Contents"><a class="docs-heading-anchor" href="#Contents">Contents</a><a id="Contents-1"></a><a class="docs-heading-anchor-permalink" href="#Contents" title="Permalink"></a></h2><ul><li><a href="index.html#LaScaMolMR">LaScaMolMR</a></li><li class="no-marker"><ul><li><a href="index.html#Overview">Overview</a></li><li><a href="index.html#Tutorial">Tutorial</a></li><li><a href="index.html#Contents">Contents</a></li><li><a href="index.html#API">API</a></li><li><a href="index.html#Index">Index</a></li></ul></li></ul><h2 id="API"><a class="docs-heading-anchor" href="#API">API</a><a id="API-1"></a><a class="docs-heading-anchor-permalink" href="#API" title="Permalink"></a></h2><h3 id="Provide-inputs"><a class="docs-heading-anchor" href="#Provide-inputs">Provide inputs</a><a id="Provide-inputs-1"></a><a class="docs-heading-anchor-permalink" href="#Provide-inputs" title="Permalink"></a></h3><p>These functions allow users to specify the input formats. The QTL summary statistics can be united in a single file or divided by exposure, chromosome or both.</p><p>For the sake of simplicity, we consider a genetic association study as a GWAS when a single phenotype was studied and as a QTLStudy otherwise.</p><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.GenVarInfo" href="#LaScaMolMR.GenVarInfo"><code>LaScaMolMR.GenVarInfo</code></a> — <span class="docstring-category">Type</span></header><section><div><p>Enum of different information present in qtl file path or inside QTL/GWAS text delimited files.</p><p><strong>values :</strong></p><pre><code class="nohighlight hljs">TRAIT_NAME  # Name of the trait (e.g. QTL protein name)
CHR         # chromosome
POS         # position in chromosome
A_EFFECT    # effect allele of SNP
A_OTHER     # reference allele of SNP
BETA        # effect size of SNP
SE          # standard error of effect size
PVAL        # Pvalue for H₀ : BETA = 0</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/inputs.jl#L15-L30">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.GWAS" href="#LaScaMolMR.GWAS"><code>LaScaMolMR.GWAS</code></a> — <span class="docstring-category">Type</span></header><section><div><p>Type GWAS which contains informations about GWAS file</p><pre><code class="nohighlight hljs">path        # path to file
columns     # Dictionary of column number =&gt; GenVarInfo type of info at column
separator   # column separator
trait_name  # name of the trait (nothing if not specified)</code></pre><p><strong>Example</strong></p><pre><code class="nohighlight hljs">gwas = GWAS(&quot;path/to/file&quot;, 
            Dict(1 =&gt; PVAL, 2 =&gt; CHR, 3 =&gt; POS, 8 =&gt; BETA, 9 =&gt; SE),
            &#39;,&#39;)</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/inputs.jl#L49-L65">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.QTLStudy" href="#LaScaMolMR.QTLStudy"><code>LaScaMolMR.QTLStudy</code></a> — <span class="docstring-category">Type</span></header><section><div><p>Type QTLStudy which contains informations about QTL file(s) format and implacement.</p><p>The method <a href="index.html#LaScaMolMR.QTLStudy_from_pattern"><code>QTLStudy_from_pattern</code></a> helps building information from patterns in file names and is the prefered methods to construct a QTLStudy.</p><p><strong>fields :</strong></p><pre><code class="nohighlight hljs">path_v                  # vector of file names (path)
traits_for_each_path    # Trait corrpespondinf to each path (nothing if files are not trait specific)
trait_v                 # names of all traits included
chr_v                   # chromosomes corresponding to each trait (nothing if not relevant)
tss_v                   # tss position corresponding to each trait (nothing if not relevant)
columns                 # Dictionary of column number =&gt; GenVarInfo type of info at column
separator               # column separator</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/inputs.jl#L78-L94">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.QTLStudy_from_pattern" href="#LaScaMolMR.QTLStudy_from_pattern"><code>LaScaMolMR.QTLStudy_from_pattern</code></a> — <span class="docstring-category">Function</span></header><section><div><p>Make QTLStudy which contains informations about QTL file(s) format and implacement from some file pattern</p><p><strong>arguments :</strong></p><p><code>folder</code> is the main folder cntaining all QTL files. <br/><code>path_pattern</code> is a vector of strngs and GenVarInfo characterizing the pattern of every file name in the folder. <br/><code>trait_v</code> is an collection of traits that should be incuded in future analysis. <br/><code>chr_v</code> is a vector of each chromosome of each QTL trait. <br/><code>tss_v</code> is a vector of every Transcription start site. <br/><code>columns</code> is a dictionary of all informations contained in columns of QTL files. <br/><code>separator</code> is the separator of QTL files. <br/><code>only_corresp_chr</code> indicates if only variants on chromosome correspoding to QTL trait should be kept. Default is <code>true</code>.      Set to <code>false</code> for Trans MR studies.</p><p>TIP : if your file contains a chromosome:position column (e.g. 1:324765), consider setting your separator to <code>[&#39;:&#39;, sep]</code></p><p><strong>Examples</strong></p><p>for a sigle file QTL :</p><pre><code class="nohighlight hljs">QTLStudy_from_pattern(&quot;some/folder&quot;, [&quot;qtl.txt&quot;], tv, cv, tssv,
                      Dict(1 =&gt; CHR, 2 =&gt; POS, 8 =&gt; PVAL, ...),
                      separator = &#39; &#39;,
                      only_corresp_chr = true)</code></pre><p>for this arhitecture (in UNIX) :</p><pre><code class="nohighlight hljs">test/data
├── exposureA
│   ├── exposureA_chr1.txt
│   └── exposureA_chr2.txt
├── exposureB
│   ├── exposureB_chr1.txt
│   └── exposureB_chr2.txt
└── exposureC
    ├── exposureC_chr3.txt
    └── exposureC_chr4.txt</code></pre><p>we get this code :</p><pre><code class="nohighlight hljs">path_pattern = [&quot;exposure&quot;, TRAIT_NAME, &quot;/exposure&quot;, TRAIT_NAME, &quot;_chr&quot;, CHR, &quot;.txt&quot;]
trait_v = [&quot;A&quot;, &quot;B&quot;, &quot;C&quot;]
chr_v = [1, 2, 3]
tss_v = [45287, 984276, 485327765]

qtl = QTLStudy_from_pattern(&quot;test/data&quot;, path_pattern, ttrait_v, chr_v, tss_v,
                            Dict(1 =&gt; CHR, 2 =&gt; POS, 8 =&gt; PVAL, ...),
                            separator = &#39; &#39;,
                            only_corresp_chr = true)</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/inputs.jl#L120-L176">source</a></section></article><h3 id="Perform-a-MR-Study"><a class="docs-heading-anchor" href="#Perform-a-MR-Study">Perform a MR Study</a><a id="Perform-a-MR-Study-1"></a><a class="docs-heading-anchor-permalink" href="#Perform-a-MR-Study" title="Permalink"></a></h3><p>The function <code>mrStudy</code> allows the users to perform meta-analysis over many pairs of traits given the input format. The function has three implementations depending on the type of studies used as exposure and outcomes. (GWAS -&gt; GWAS, QTL -&gt; GWAS, GWAS -&gt; QTL)</p><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.mrStudy" href="#LaScaMolMR.mrStudy"><code>LaScaMolMR.mrStudy</code></a> — <span class="docstring-category">Function</span></header><section><div><p>Perform a Mendelian Randomization study with exposure QTL and outcome GWAS</p><p><strong>arguments :</strong></p><p><code>exposure::QTLStudy</code> : exposure QTL data <br/><code>outcome::GWAS</code> : outcome gas data <br/><code>type::AbstractString</code> : specifies if the analysis is a trans or cis MR analysis (is either <code>&quot;trans&quot;</code> or <code>&quot;cis&quot;</code>)<br/><code>bedbimfam_dirnames::AbstractArray{&lt;:AbstractString}</code> : base names of Plink bedbimfam files for reference genotypes      (see <a href="https://openmendel.github.io/SnpArrays.jl/latest/">SnpArrays documentation</a>)<br/> <strong>options :</strong><br/> <code>approach::String</code>: name of MR study aproach chosen (either naive, test or MiLoP) (default is &quot;naive&quot;)<br/><code>p_tresh::Float64</code>: pvalue threshold for a SNP to be considered associated to an exposure (default is 1e-3)<br/><code>window::Integer</code>: maximal distance between a potential Instrument Variable and transciption start site of gene exposure (default is 500<em>000)<br/>`r2</em>tresh::Float64`: maximial corrlation between to SNPs (default is 0.1)<br/><code>exposure_filtered::Bool</code> : If true, the exposure files are considered already filtered will not filtered      on distance to tss and level of significance (default is false)<br/><code>mr_methods::AbstractVector{Function}</code> : Functions to use to estimate effect of exposure on outcome.     Any Function taking four vectors of same length (βoutcome, se<em>outcome, βexposure, se</em>exposure) and a Float (α)      and returns a value of type <a href="index.html#LaScaMolMR.mr_output"><code>mr_output</code></a> can be used, that includes user defined functions.      Functions already implemented in this module include <a href="index.html#LaScaMolMR.mr_ivw"><code>mr_ivw</code></a>, <a href="index.html#LaScaMolMR.mr_egger"><code>mr_egger</code></a>, <a href="index.html#LaScaMolMR.mr_wm"><code>mr_wm</code></a> and <a href="index.html#LaScaMolMR.mr_wald"><code>mr_wald</code></a>. default value is <code>[mr_ivw, mr_egger]</code> <br/><code>α::Float64</code> : α value for confidance intervals of parameter estimations in MR (e.g. 95% CI is α = 0.05, which is the default value)<br/><code>trsf_pval_exp::Union{Function, Nothing}</code> : Transformation to apply to pvalues in exposure dataset<br/><code>trsf_pval_out::Union{Function, Nothing}</code> : t = Transormation to apply on pvalues in outcome dataset<br/><code>low_ram::Bool</code> : If true, if the exposure files each contain only one exposure trait, <a href="index.html#LaScaMolMR.mrStudyNFolds"><code>mrStudyNFolds</code></a> with <code>n_folds</code> of 10 will be used.<br/><code>write_ivs::AbstractString</code> : write selected Instrumental variables to specified directory<br/><code>write_filtered_exposure::AbstractString</code> : write a filtered version of exposure files to specified file name.     This file will be tab separated and will only contain columns necessary for further MR Studies.<br/><code>filter_beta_raio::Real</code> : Filter IVs for which the exposure effect is <code>filter_beta_raio</code> times outcome effect or greater. default is 0.<br/><code>pval_bigfloat::Bool</code> : use <code>true</code> if pvalues can be under <code>5e-324</code>. (default is <code>false</code>)<br/><code>min_maf::Real</code> : minimal variant maf to be kept as a potential IV. (default is 0)<br/><code>infos::Bool</code> : If true, infos about advancement compute are printed to terminal (default is <code>true</code>)</p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">results = mrStudy(qtl, gwas, &quot;trans&quot;, genotypes, 10, approach = &quot;naive&quot;, window = 250000, trsf_pval_exp = x -&gt; exp10.(x))</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/mrStudy.jl#L410-L450">source</a></section><section><div><p>Perform a Trans-Mendelian Randomization study with exposure GWAS and outcome GWAS</p><p><strong>arguments :</strong></p><p><code>exposure::GWAS</code> : exposure QTL data <br/><code>outcome::GWAS</code> : outcome GWAS data <br/><code>bedbimfam_dirnames::AbstractArray{&lt;:AbstractString}</code> : base names of Plink bedbimfam files for reference genotypes      (see <a href="https://openmendel.github.io/SnpArrays.jl/latest/">SnpArrays documentation</a>)<br/> <strong>options :</strong> <br/> <code>approach::String</code>: name of MR study aproach chosen (either naive, test or MiLoP) (default is &quot;naive&quot;)<br/><code>p_tresh::Float64</code>: pvalue threshold for a SNP to be considered associated to an exposure (default is 1e-3)<br/><code>r2_tresh::Float64</code>: maximial corrlation between to SNPs (default is 0.1)<br/><code>exposure_filtered::Bool</code> : If true, the exposure files are considered already filtered will not filtered      on distance to tss and level of significance (default is false)<br/><code>mr_methods::AbstractVector{Function}</code> : Functions to use to estimate effect of exposure on outcome.     Any Function taking four vectors of same length (βoutcome, se<em>outcome, βexposure, se</em>exposure) and a Float (α)      and returns a value of type <a href="index.html#LaScaMolMR.mr_output"><code>mr_output</code></a> can be used, that includes user defined functions.      Functions already implemented in this module include <a href="index.html#LaScaMolMR.mr_ivw"><code>mr_ivw</code></a>, <a href="index.html#LaScaMolMR.mr_egger"><code>mr_egger</code></a>, <a href="index.html#LaScaMolMR.mr_wm"><code>mr_wm</code></a> and <a href="index.html#LaScaMolMR.mr_wald"><code>mr_wald</code></a>. default value is <code>[mr_ivw, mr_egger]</code> <br/><code>α::Float64</code> : α value for confidance intervals of parameter estimations in MR (e.g. 95% CI is α = 0.05, which is the default value)<br/><code>trsf_pval_exp::Union{Function, Nothing}</code> : Transformation to apply to pvalues in exposure dataset<br/><code>trsf_pval_out::Union{Function, Nothing}</code> : t = Transormation to apply on pvalues in outcome dataset<br/><code>low_ram::Bool</code> : If true, if the exposure files each contain only one exposure trait, <a href="index.html#LaScaMolMR.mrStudyNFolds"><code>mrStudyNFolds</code></a> with <code>n_folds</code> of 10 will be used. <code>write_ivs::AbstractString</code> : write selected Instrumental variables to specified directory<br/><code>write_filtered_exposure::AbstractString</code> : write a filtered version of exposure files to specified file name.     This file will be tab separated and will only contain columns necessary for further MR Studies.<br/><code>pval_bigfloat::Bool</code> : use <code>true</code> if pvalues can be under <code>5e-324</code>. (default is <code>false</code>)<br/><code>filter_beta_raio::Real</code> : Filter IVs for which the exposure effect is <code>filter_beta_raio</code> times outcome effect or greater. default is 0.<br/><code>infos::Bool</code> : If true, infos about advancement compute are printed to terminal (default is <code>true</code>)</p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">results = mrStudy(gwas1, gwas2, genotypes, 10, approach = &quot;naive&quot;, trsf_pval_exp = x -&gt; exp10.(x))</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/mrStudy.jl#L591-L628">source</a></section><section><div><p>Perform a Trans-Mendelian Randomization study with exposure GWAS and outcome QTL</p><p><strong>arguments :</strong></p><p><code>exposure::GWAS</code> : exposure GWAS data <br/><code>outcome::QTLStudy</code> : outcome QTL data <br/><code>bedbimfam_dirnames::AbstractArray{&lt;:AbstractString}</code> : base names of Plink bedbimfam files for reference genotypes      (see <a href="https://openmendel.github.io/SnpArrays.jl/latest/">SnpArrays documentation</a>)<br/> <strong>options :</strong> </p><p><code>approach::String</code>: name of MR study aproach chosen (either naive, test or MiLoP) (default is &quot;naive&quot;)<br/><code>p_tresh::Float64</code>: pvalue threshold for a SNP to be considered associated to an exposure (default is 1e-3)<br/><code>r2_tresh::Float64</code>: maximial corrlation between to SNPs (default is 0.1)<br/><code>mr_methods::AbstractVector{Function}</code> : Functions to use to estimate effect of exposure on outcome.     Any Function taking four vectors of same length (βoutcome, se<em>outcome, βexposure, se</em>exposure) and a Float (α)      and returns a value of type <a href="index.html#LaScaMolMR.mr_output"><code>mr_output</code></a> can be used, that includes user defined functions.      Functions already implemented in this module include <a href="index.html#LaScaMolMR.mr_ivw"><code>mr_ivw</code></a>, <a href="index.html#LaScaMolMR.mr_egger"><code>mr_egger</code></a>, <a href="index.html#LaScaMolMR.mr_wm"><code>mr_wm</code></a> and <a href="index.html#LaScaMolMR.mr_wald"><code>mr_wald</code></a>. default value is <code>[mr_ivw, mr_egger]</code> <br/><code>α::Float64</code> : α value for confidance intervals of parameter estimations in MR (e.g. 95% CI is α = 0.05, which is the default value)<br/><code>trsf_pval_exp::Union{Function, Nothing}</code> : Transformation to apply to pvalues in exposure dataset<br/><code>trsf_pval_out::Union{Function, Nothing}</code> : t = Transormation to apply on pvalues in outcome dataset<br/><code>low_ram::Bool</code> : If true, if the exposure files each contain only one exposure trait, <a href="index.html#LaScaMolMR.mrStudyNFolds"><code>mrStudyNFolds</code></a> with <code>n_folds</code> of 10 will be used. <code>write_ivs::AbstractString</code> : write selected Instrumental variables to specified directory<br/><code>pval_bigfloat::Bool</code> : use <code>true</code> if pvalues can be under <code>5e-324</code>. (default is <code>false</code>)<br/><code>filter_beta_raio::Real</code> : Filter IVs for which the exposure effect is <code>filter_beta_raio</code> times outcome effect or greater. default is 0.<br/><code>infos::Bool</code> : If true, infos about advancement compute are printed to terminal (default is <code>true</code>)</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/mrStudy.jl#L675-L703">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.mrStudyNFolds" href="#LaScaMolMR.mrStudyNFolds"><code>LaScaMolMR.mrStudyNFolds</code></a> — <span class="docstring-category">Function</span></header><section><div><p>Perform Mendelian Randomization study from QTL to GWAS and separating exposure in n folds to avoid loading full dataset in memory.      (works if separated in multiple files only)</p><p><strong>arguments :</strong></p><p><code>exposure::QTLStudy</code> : exposure QTL data <br/><code>outcome::GWAS</code> : outcome GWAS data <br/><code>type::AbstractString</code> : specifies if the analysis is a trans or cis MR analysis (is either <code>&quot;trans&quot;</code> or <code>&quot;cis&quot;</code>)<br/><code>bedbimfam_dirnames::AbstractArray{&lt;:AbstractString}</code> : base names of Plink bedbimfam files for reference genotypes      (see <a href="https://openmendel.github.io/SnpArrays.jl/latest/">SnpArrays documentation</a>)<br/></p><p><strong>options :</strong></p><p>mrStudyNFolds contains the same options as <a href="index.html#LaScaMolMR.mrStudy"><code>mrStudy</code></a><br/><code>n_folds::Integer</code> : Number f folds to separate QTl into.</p><p>see <a href="index.html#LaScaMolMR.mrStudy"><code>mrStudy</code></a> for other options and examples.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/mrStudy.jl#L271-L290">source</a></section><section><div><p>Perform Mendelian Randomization study from GWAS to QTL and separating outcome in n folds to avoid loading full dataset in memory.      (works if separated in multiple files only)</p><p><strong>arguments :</strong></p><p><code>exposure::GWAS</code> : exposure GWAS data <br/><code>outcome::QTLStudy</code> : outcome QTL data <br/><code>bedbimfam_dirnames::AbstractArray{&lt;:AbstractString}</code> : base names of Plink bedbimfam files for reference genotypes      (see <a href="https://openmendel.github.io/SnpArrays.jl/latest/">SnpArrays documentation</a>)<br/></p><p><strong>options :</strong></p><p>mrStudyNFolds contains the same options as <a href="index.html#LaScaMolMR.mrStudy"><code>mrStudy</code></a><br/><code>n_folds::Integer</code> : Number f folds to separate QTl into.</p><p>see <a href="index.html#LaScaMolMR.mrStudy"><code>mrStudy</code></a> for other options and examples.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/mrStudy.jl#L343-L361">source</a></section></article><h3 id="Linkage-Desiquilibrium-and-MR"><a class="docs-heading-anchor" href="#Linkage-Desiquilibrium-and-MR">Linkage Desiquilibrium and MR</a><a id="Linkage-Desiquilibrium-and-MR-1"></a><a class="docs-heading-anchor-permalink" href="#Linkage-Desiquilibrium-and-MR" title="Permalink"></a></h3><p>The <code>ClumpAndMR</code> function performs clumping over every pair of (exposure, outcome) and calls Mendelian randomization fonctions. You can thus use it if potential IV selection was already performed.</p><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.ClumpAndMR" href="#LaScaMolMR.ClumpAndMR"><code>LaScaMolMR.ClumpAndMR</code></a> — <span class="docstring-category">Function</span></header><section><div><p>Keep independant IVs and perform MR in Omic-wide MR (Trans or Cis)     Returns a Dataset of results for each exposure (rows).</p><p><strong>arguments :</strong> </p><p><code>data</code> : grouped dataset (exposure trait) with specific column names :      <code>:trait</code>, <code>:chr</code>, <code>:pos</code>, <code>:β_exp</code>, <code>β_out</code>, <code>:se_exp</code>, <code>:se_out</code> among others. <br/><code>GenotypesArr::AbstractVector{SnpData}</code> : Reference genotypes. (see <a href="https://github.com/OpenMendel/SnpArrays.jl"><code>SnpArrays</code></a>) formated with <a href="index.html#LaScaMolMR.formatSnpData!"><code>formatSnpData!</code></a>.</p><p><strong>options :</strong></p><p><code>one_file_per_chr_plink::Bool</code> : If true, it is considered that the each index in GenotypesArr corresponds to a chromosome number (default is true) <br/><code>r2_tresh::AbstractFloat</code> : The maximal r² correlation coefficient for two snps to be considered independant. (default is 0.1) <br/><code>mr_methods::AbstractVector{Function}</code>: Functions to use to estimate effect of exposure on outcome.     Any Function taking four vectors of same length (βoutcome, se<em>outcome, βexposure, se</em>exposure) and a Float (α)      and returns a value of type <a href="index.html#LaScaMolMR.mr_output"><code>mr_output</code></a> can be used, that includes user defined functions.      Functions already implemented in this module include <a href="index.html#LaScaMolMR.mr_ivw"><code>mr_ivw</code></a>, <a href="index.html#LaScaMolMR.mr_egger"><code>mr_egger</code></a>, <a href="index.html#LaScaMolMR.mr_wm"><code>mr_wm</code></a>     and <a href="index.html#LaScaMolMR.mr_wald"><code>mr_wald</code></a>. default value is <code>[mr_ivw, mr_egger]</code> <br/><code>α:AbstractFloat</code> : α value for confidance intervals of parameter estimations in MR (e.g. 95% CI is α = 0.05, which is the default value)<br/><code>write_ivs::AbstractString</code> : write selected Instrumental variables to specified directory</p><p><strong>Examples :</strong></p><pre><code class="language-julia hljs">res_d = ClumpAndMR(d, GenotypesArr, r2_tresh = 0.1)</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/ClumpAndMR.jl#L22-L49">source</a></section></article><h3 id="Mendelian-Randomization-functions"><a class="docs-heading-anchor" href="#Mendelian-Randomization-functions">Mendelian Randomization functions</a><a id="Mendelian-Randomization-functions-1"></a><a class="docs-heading-anchor-permalink" href="#Mendelian-Randomization-functions" title="Permalink"></a></h3><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.mr_output" href="#LaScaMolMR.mr_output"><code>LaScaMolMR.mr_output</code></a> — <span class="docstring-category">Type</span></header><section><div><p>Struct encapsulating the outputs of a Mendelian Randomization analysis</p><p>fields :</p><p><code>nivs</code> : number of ivs included in regression <br/><code>effect</code> : effect size estimate ̂γ₀ <br/><code>se_effect</code> : standard error of effect size estimate <br/><code>ci_low</code> : lower bound of confidence interval for effect size <br/><code>ci_high</code> : higher bound of confidence interval for effect size <br/><code>p</code> : effect size p-value for H₀ : γ₀ = 0 <br/><code>intercept</code> : intercept estimate ̂γ₁ <br/><code>p_intercept</code> : p-value for H₀ : γ₁ = 0 <br/><code>ci_low_intercept</code> : lower bound of confidence interval for intercept <br/><code>ci_high_intercept</code> : higher bound of confidence interval for intercept <br/><code>heter_stat</code> : heterogenetity statistic corresponding to Cochran ̂t <br/><code>heter_p</code> : heterogeneity p-value for H₀ : t = 0</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/mrPerf.jl#L59-L76">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.mr_wald" href="#LaScaMolMR.mr_wald"><code>LaScaMolMR.mr_wald</code></a> — <span class="docstring-category">Function</span></header><section><div><p>Wald ratio for Mendelian Randomization with a single instrumental variable      y is outcome, x is exposure</p><p><strong>arguments :</strong></p><p><code>β_Y</code> : outcome effect sizes</p><p><code>se_β_Y</code> : standard error of outcome effect size</p><p><code>β_X</code> : exposure effect size</p><p><code>α</code> : α value for confidence intervals (default is 0.05)</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/mrPerf.jl#L111-L125">source</a></section><section><div><p>Wald ratio for Mendelian Randomization with a single instrumental variable      y is outcome, x is exposure</p><p><strong>arguments :</strong></p><p><code>β_Y</code> : vector of outcome effect sizes</p><p><code>se_β_Y</code> : vector of standard error of outcome effect sizes</p><p><code>β_X</code> : vector of exposure effect sizes</p><p><code>se_β_X</code> : vector of standard error for exposure effect sizes</p><p><code>α</code> : α value for confidence intervals (default is 0.05)</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/mrPerf.jl#L140-L156">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.mr_ivw" href="#LaScaMolMR.mr_ivw"><code>LaScaMolMR.mr_ivw</code></a> — <span class="docstring-category">Function</span></header><section><div><p>Inverse variance weighted linear regression with simple weights (se(B_Y)^-2) Mendelian Randomization     Y is outcome, X is exposure</p><p>currently waiting for GLM.jl PR#487 to be merged to use analytical weights instead of doing calculations twice...</p><p><strong>arguments :</strong></p><p><code>β_Y</code> : vector of outcome effect sizes</p><p><code>se_β_Y</code> : vector of standard error of outcome effect sizes</p><p><code>β_X</code> : vector of exposure effect sizes</p><p><code>se_β_X</code> : vector of standard error for exposure effect sizes</p><p><code>α</code> : α value for confidence intervals (default is 0.05)</p><p><strong>options :</strong></p><p><code>model::Symbol</code> : <code>:random</code>, <code>:fixed</code> or <code>:default</code> effect type. <code>:default</code> is fixed if less than 4 ivs, random otherwise.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/mrPerf.jl#L166-L188">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.mr_egger" href="#LaScaMolMR.mr_egger"><code>LaScaMolMR.mr_egger</code></a> — <span class="docstring-category">Function</span></header><section><div><p>Egger Mendelian Randomization     Y is outcome, X is exposure</p><p>currently waiting for GLM.jl PR#487 to be merged to use analytical weights instead of doing calculations twice...</p><p><strong>arguments :</strong></p><p><code>β_Y</code> : vector of outcome effect sizes</p><p><code>se_β_Y</code> : vector of standard error of outcome effect sizes</p><p><code>β_X</code> : vector of exposure effect sizes</p><p><code>se_β_X</code> : vector of standard error for exposure effect sizes</p><p><code>α</code> : α value for confidence intervals (default is 0.05)</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/mrPerf.jl#L243-L261">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.mr_wm" href="#LaScaMolMR.mr_wm"><code>LaScaMolMR.mr_wm</code></a> — <span class="docstring-category">Function</span></header><section><div><p>Weighted Median Mendelian Randomization (Bowden et al., 2015)     Y is outcome, X is exposure.</p><p><strong>arguments :</strong></p><p><code>β_Y</code> : vector of outcome effect sizes</p><p><code>se_β_Y</code> : vector of standard error of outcome effect sizes</p><p><code>β_X</code> : vector of exposure effect sizes</p><p><code>se_β_X</code> : vector of standard error for exposure effect sizes</p><p><code>α</code> : α value for confidence intervals (default is 0.05)</p><p><strong>options :</strong></p><p><code>iterations</code> : number of iterations for estimate of standrard error or effect size. <code>seed</code> : seed of random generator</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/mrPerf.jl#L314-L334">source</a></section></article><h3 id="Exported-Utilities"><a class="docs-heading-anchor" href="#Exported-Utilities">Exported Utilities</a><a id="Exported-Utilities-1"></a><a class="docs-heading-anchor-permalink" href="#Exported-Utilities" title="Permalink"></a></h3><h4 id="Inputs"><a class="docs-heading-anchor" href="#Inputs">Inputs</a><a id="Inputs-1"></a><a class="docs-heading-anchor-permalink" href="#Inputs" title="Permalink"></a></h4><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.nfolds" href="#LaScaMolMR.nfolds"><code>LaScaMolMR.nfolds</code></a> — <span class="docstring-category">Function</span></header><section><div><p>Partitionate QTLStudy in n folds. Returns a vector of QTLStudy in which each element contains a subsets of file paths and corresponding traits. </p><p><strong>arguments :</strong></p><p><code>x::QTLStudy</code> : the qtl files and informations (see <a href="index.html#LaScaMolMR.QTLStudy"><code>QTLStudy</code></a>) <br/><code>m::Integer</code> : the number of folds</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/inputs.jl#L297-L304">source</a></section></article><h4 id="LD"><a class="docs-heading-anchor" href="#LD">LD</a><a id="LD-1"></a><a class="docs-heading-anchor-permalink" href="#LD" title="Permalink"></a></h4><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.formatSnpData!" href="#LaScaMolMR.formatSnpData!"><code>LaScaMolMR.formatSnpData!</code></a> — <span class="docstring-category">Function</span></header><section><div><p>format Genotype information contained in SnpData for optimised snp search based on chromosome position. Adds a column of tuple (chr::Int8, pos::Int) in snp<em>info and sorts snp</em>info accordingly.     returns nothing</p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">ref = SnpData(datadir(&quot;some/data&quot;))

formatSnpData!(ref)</code></pre><p>Note this function does not take into account the user might want to write SnpData in bed, bim, fam format.      The changes done by this function can and will cause problems if writing plink files after calling <code>formatSnpData!</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/ld.jl#L215-L230">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="LaScaMolMR.clump" href="#LaScaMolMR.clump"><code>LaScaMolMR.clump</code></a> — <span class="docstring-category">Function</span></header><section><div><p>threaded implementation of the clumping algorithm prioritising first snps in given Vector and formated Genotypes SnpData (see <a href="index.html#LaScaMolMR.formatSnpData!"><code>formatSnpData!</code></a>)     returns a vector of boolean indicating if each given snp is kept</p><p>arguments :</p><p>ref_genotypes : reference SnpData (<a href="https://github.com/OpenMendel/SnpArrays.jl">SnpArrays</a>) <br/>snps : vector of chromosome position tuple for each variant</p><p>options : </p><p><code>formated</code> : indicates if ref SnpData is already formated according to :chr<em>pos (chr, pos)<br/>`r2</em>tresh` : minimal r² for 2 snps to be considered strongly correlated.</p><p><strong>Examples :</strong></p><pre><code class="language-julia hljs">julia&gt; ref = SnpData(datadir(&quot;some/data&quot;));

julia&gt; kept_v_b::Vector{Bool} = clump([(1, 123), (1, 456), (1, 789)], ref)
3-element Vector{Int64}:
 1
 0
 1

julia&gt; formatSnpData!(ref);

julia&gt; kept_v_b::Vector{Bool} = clump([(1, 123), (1, 456), (1, 789)], ref, formated = true)
3-element Vector{Int64}:
 1
 0
 1</code></pre><p>If formatSnpData has already been called on good snp info type (<code>:chr_pos</code> or <code>:snpid</code>), <code>formated = true</code> option does not verify or modify formating. See <a href="index.html#LaScaMolMR.formatSnpData!"><code>formatSnpData!</code></a>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/SamuelMathieu-code/LaScaMolMR.jl/blob/8dbf84c4a983a3f72e8195680bb144e49bca1d27/src/ld.jl#L128-L164">source</a></section></article><h2 id="Index"><a class="docs-heading-anchor" href="#Index">Index</a><a id="Index-1"></a><a class="docs-heading-anchor-permalink" href="#Index" title="Permalink"></a></h2><ul><li><a href="index.html#LaScaMolMR.GWAS"><code>LaScaMolMR.GWAS</code></a></li><li><a href="index.html#LaScaMolMR.GenVarInfo"><code>LaScaMolMR.GenVarInfo</code></a></li><li><a href="index.html#LaScaMolMR.QTLStudy"><code>LaScaMolMR.QTLStudy</code></a></li><li><a href="index.html#LaScaMolMR.mr_output"><code>LaScaMolMR.mr_output</code></a></li><li><a href="index.html#LaScaMolMR.ClumpAndMR"><code>LaScaMolMR.ClumpAndMR</code></a></li><li><a href="index.html#LaScaMolMR.QTLStudy_from_pattern"><code>LaScaMolMR.QTLStudy_from_pattern</code></a></li><li><a href="index.html#LaScaMolMR.clump"><code>LaScaMolMR.clump</code></a></li><li><a href="index.html#LaScaMolMR.formatSnpData!"><code>LaScaMolMR.formatSnpData!</code></a></li><li><a href="index.html#LaScaMolMR.mrStudy"><code>LaScaMolMR.mrStudy</code></a></li><li><a href="index.html#LaScaMolMR.mrStudyNFolds"><code>LaScaMolMR.mrStudyNFolds</code></a></li><li><a href="index.html#LaScaMolMR.mr_egger"><code>LaScaMolMR.mr_egger</code></a></li><li><a href="index.html#LaScaMolMR.mr_ivw"><code>LaScaMolMR.mr_ivw</code></a></li><li><a href="index.html#LaScaMolMR.mr_wald"><code>LaScaMolMR.mr_wald</code></a></li><li><a href="index.html#LaScaMolMR.mr_wm"><code>LaScaMolMR.mr_wm</code></a></li><li><a href="index.html#LaScaMolMR.nfolds"><code>LaScaMolMR.nfolds</code></a></li></ul><h3 id="Authors"><a class="docs-heading-anchor" href="#Authors">Authors</a><a id="Authors-1"></a><a class="docs-heading-anchor-permalink" href="#Authors" title="Permalink"></a></h3><p><strong>Samuel Mathieu</strong>, Hippolyte Minvielle Moncla, Mewen Briend, Valentine Duclos, Anne Rufiange, Patrick Mathieu</p></article><nav class="docs-footer"><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="auto">Automatic (OS)</option><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option><option value="catppuccin-latte">catppuccin-latte</option><option value="catppuccin-frappe">catppuccin-frappe</option><option value="catppuccin-macchiato">catppuccin-macchiato</option><option value="catppuccin-mocha">catppuccin-mocha</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 1.5.0 on <span class="colophon-date" title="Tuesday 2 July 2024 16:29">Tuesday 2 July 2024</span>. Using Julia version 1.10.4.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
